fossunited/setup.py

```diff
@@ -0,0 +1,109 @@
+import frappe
+from frappe.custom.doctype.custom_field.custom_field import create_custom_fields
+
+
+def after_install():
+    create_custom_fields(get_custom_fields(), ignore_validate=True)
+
+
+def before_uninstall():
+    delete_custom_fields(get_custom_fields())
+
+
+def delete_custom_fields(custom_fields: dict):
+    """
+    :param custom_fields: a dict like `{'Salary Slip': [{fieldname: 'loans', ...}]}`
+    """
+    for doctype, fields in custom_fields.items():
+        frappe.db.delete(
+            "Custom Field",
+            {
+                "fieldname": ("in", [field["fieldname"] for field in fields]),
+                "dt": doctype,
+            },
+        )
+
+        frappe.clear_cache(doctype=doctype)
+
+
+def get_custom_fields():
+    """
+    FOSSUnited specific custom fields that need to be added to base Frappe doctypes
```

Comment:
q. why do these custom fields need to be added to the base doctypes? we didn't have to do this earlier so why now?

Reply:
Newsletter is  a core frappe doctype. We are customizing this core doctype at the installation of our app, hence the `before_install` and `after_install` hooks earlier. 

We are doing this, because there are a ton of functionalities included already in the core [Newsletter doctype,](https://github.com/frappe/frappe/blob/develop/frappe/email/doctype/newsletter/newsletter.py) so rather than rewriting that code, the more efficient way is to add these 2 custom fields and use the newsletter doctype as we want.

We never had to do this earlier because there was no use case for it before.